### DISCUSSION TITLE
Is Adam the end of Gradient descent (and model Optimizers)?

### DISCUSSION THREAD

**Discussion by BradKML** [keras-team/autokeras#1937] [Ideas]
Currently it is SGD and Adam that are the most commonly used (and sometimes AdamW for newer applications), but there are things like Nadam, Nesterov, RMSProp, Momentum, AdaDelta, LARS, AdaGrad, AdaMax, Sophia, AdaBound, AdaFactor, AdaLomo...
Sadly not everything is included for TensorFlow (and some are in PyTorch so IDK)
Reference https://github.com/keras-team/autokeras/blob/master/autokeras/graph.py https://www.tensorflow.org/api_docs/python/tf/keras/optimizers
